feat: add support `STM32WLE5CCUx`

elpiel opened the pull request:
The primary usage is to support the `STM32WLE5CCU6` and in particular this board: https://www.olimex.com/Products/IoT/LoRa/BB-STM32WL/ with it's the Dev board - https://www.olimex.com/Products/IoT/LoRa/LoRa-STM32WL-DevKit

I'm not 100% sure about the flashing algorithm and if I've missed anything else - let me know.

bugadani commented:
> I'm not 100% sure about the flashing algorithm

Assuming you tried this, because you need this, did it work? Considering that CCUx seem to only differ in pin count/packaging from JCUx, if you could add a changelog, I think we could merge this.

elpiel commented:
@bugadani yes it is tested and I successfully flashed a `lora-rs` example.
I initially tried with `stm32wlexx_128` algorithm but it didn't work then saw it the datasheet mentioning CM4 and switched to `stm32wlxx_cm4`

I've updated the Changelog, it's ready for merge :)

bugadani commented:
Thanks, although probe-rs (and its CI) need the changelog entry [a bit differently](https://github.com/probe-rs/probe-rs/blob/master/CONTRIBUTING.md#changelog-entries).

elpiel commented:
> Thanks, although probe-rs (and its CI) need the changelog entry [a bit differently](https://github.com/probe-rs/probe-rs/blob/master/CONTRIBUTING.md#changelog-entries).

Thanks! I updated my PR, now the Changelog check passes.

bugadani reviewed (approved):
Thank you!
